cmake/CPM.cmake

```
@@ -269,10 +269,21 @@ endfunction()
 # finding the system library
 function(cpm_create_module_file Name)
   if(NOT CPM_DONT_UPDATE_MODULE_PATH)
-    # erase any previous modules
-    file(WRITE ${CPM_MODULE_PATH}/Find${Name}.cmake
-         "include(\"${CPM_FILE}\")\n${ARGN}\nset(${Name}_FOUND TRUE)"
-    )
+    if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.24.0")
+      string(TOLOWER ${Name} NameLower)
```

Would it make sense to add a comment linking to the CMake docs here, so that others can quickly understand what this code does?

```suggestion
      # the following is used for integration with FindPackage
      # https://cmake.org/cmake/help/latest/module/FetchContent.html#fetchcontent-find-package-integration-examples
      string(TOLOWER ${Name} NameLower)
```